### ISSUE TITLE
Nondifferentiable likelihood?

### ISSUE THREAD

**Issue by floswald** [tpapp/DynamicHMC.jl#130]
Hey there 

I wanted to ask whether it's feasible to optimise a nondifferentiable posterior. For large macro models it's often impossible to derive that gradient, either manually or via AD. Like if there is simulation involved to compute moments which go into the likelihood. I have a simple package at floswald/SMM.jl but I'd like to leverage your structure. In particular diagnostics. Is there any hope you think? Thanks for the great work, here and elsewhere.
F

**Comment by tpapp** [comment]
Yes, I am working on exactly this. Should have updates in a month or so. If you are interested, I can share a preliminary working paper then.

**Comment by floswald** [comment]
Brilliant. Yes much interested! Also can contribute some (at least testing?)

**Comment by Closed-Limelike-Curves** [comment]
Was this ever implemented?

**Comment by tpapp** [comment]
Still WIP.

**Comment by floswald** [comment]
do you have strategy to implement it already? I know it takes extra time to explain to outsiders, but maybe if you point us to the main concerned parts and what's the overall strategy we could try and help? just a thought

**Comment by tpapp** [comment]
Sorry, COVID-related stuff delayed this project a bit, so I am still cleaning up things for a paper.

What I mostly have is a bag of various tricks that worked for some macro projects, including smaller DSGE and hetag models. Currently I am experimenting with how they scale to bigger ones. 

It would be easier to be specific about concrete model, feel free to contact me in private and I am happy to help (or cooperate on a project, if it gets very involved).
